hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/catalyst/catalog/HoodieCatalogTable.scala

```diff
@@ -171,9 +175,14 @@ class HoodieCatalogTable(val spark: SparkSession, val table: CatalogTable) exten
         .setTableCreateSchema(SchemaConverters.toAvroType(finalSchema).toString())
         .initTable(hadoopConf, tableLocation)
     } else {
+      val hoodieDatabaseName = if (hoodieTableExists) databaseName else
+        formatName(spark, table.identifier.database.getOrElse(spark.sessionState.catalog.getCurrentDatabase))
+      val hoodieTableName = if (hoodieTableExists) tableName else table.identifier.table
```

Comment:
here you don't need to judge `hoodieTableExists` is true or not. it must be false.

Reply:
OK, it has been modified